community/cypher/cypher/src/test/scala/org/neo4j/cypher/KeysAcceptanceTest.scala

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher
+
+import internal.helpers.CollectionSupport
+import org.scalatest.Assertions
+import org.neo4j.graphdb.{Relationship, Node}
+import org.scalautils.LegacyTripleEquals
+
+class KeysAcceptanceTest extends ExecutionEngineFunSuite  with QueryStatisticsTestSupport {
```

Comment:
It would be lovely, for the sake of completeness, to add a couple of tests where the node/rel is null (using `OPTIONAL MATCH`), and maybe a couple of expected failure situations and how the function is expected to handle them?